sync-core/src/main/java/com/cloudant/sync/query/UnindexedMatcher.java

```diff
@@ -233,17 +234,23 @@ private boolean executeSelectorTree(QueryNode node, DocumentRevision rev) {
                 operator = (String) operatorExpression.keySet().toArray()[0];
             }
 
-            Object expected = operatorExpression.get(operator);
+            Object expectedValues = operatorExpression.get(operator);
+            if (!(expectedValues instanceof List)) {
+                expectedValues = Arrays.asList(expectedValues);
+            }
```

Comment:
Might be worth a note here that you are doing this so $in can be treated like $eq and we don't have to special case $in (which I like).